design_docs/generalize_calendar.md

```diff
@@ -0,0 +1,249 @@
+are converted to `numpy.datetime64` objects at various stages when using 
+features from pandas, raising errors when dates are out of range.  While an
+alternative date class (e.g. `netcdftime.DatetimNoLeap`) might be used to
+represent dates on the 'gregorian_noleap' calendar, there is no such
+preexisting alternative for the 'gregorian' calendar.
+
+The solution proposed herein is to store time as floating-point days since the
+reference date 0001-01-01 and to convert dates in this format to 
+`datetime.datetime` and `MpasRelativeDelta` objects whenever mathematical
+manipulation of dates is required.
+
+A successful implementation would produce essentially identical analysis to
+what is currently produced, but making use of the dates from the MPAS calendar
+(whether Gregorian or 365-day) without the need for artifical offsets (e.g.
+`yearOffset` used in the current code.  Plots of horizontal fields would remain
+unchanged while plots of time series would have a time axis with the simulation
+date instead of the offset date.
+
+
+<h1> Requirements </h1>
+
+<h2> Requirement: The 'Time' coordinate of xarray data sets must be consistent 
+with the MPAS calendar <br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+For all data sets used in the analysis, the 'Time' coordinate must represent dates 
+on the appropriate MPAS calendar, either 'gregorain' or 'gregorain_noleap', depending
+on the namelist option 'config_calendar_type'.  There must be ways of mathematically
+manipulating times (e.g. adding/subtracting offsets and figuring out the amount of time
+between two dates) and of making plots that are consistent with these calendars.
+
+<h2> Requirement: The 'Time' coordinate of xarray data sets must support years 
+0001 and 9999, and value, both positive and negative<br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+For all data sets used in the analysis, the 'Time' coordinate must, at a minimum, 
+support years between 0001 and 9999 (the range of `datetime.datetim`) and preferably
+a broader range.
+
+
+<h1> Algorithmic Formulations (optional) </h1>
+
+<h2> Design solution: The 'Time' coordinate of xarray data sets must be consistent
+with the MPAS calendar <br>
+Date last modified: 2017/02/09 <br>
+Contributors: Xylar Asay-Davis
+</h2>
+
+The proposed solution represents time in `xarray.DataSet` objects as the number of
+days since the reference date 0001-01-01.  For purposes of performing mathematical
+operations and plotting dates, these values will be converted to 
+`datetime.datetime` objects (via the proposed `daysToDatetime` utility function)
+and back (via the proposed `datetimeToDays`).
+
+The conversion operations will be performed with the calendar-aware functions
+`netCDF4.date2num` and `netCDF4.num2date` and will support lists/arrays of dates
+(for efficiency and simplicity of calling code) in addition to single values.
```

Comment:
No, this is still the solution.  These function are just wrapped in the `datetimeToDays` and `daysToDatetime` functions for simplicity.  I will clarify.